### DISCUSSION TITLE
Query with ickle using field wildcards?

### DISCUSSION THREAD

**Discussion by StephenOTT** [infinispan/infinispan#12081] [Q&A]
Is it possible to using partial field names / wildcards for a ickle query? Examples show wildcards for field values but can't find anything on field names. 

Thanks

**Comment by fax4ever** [comment]
Hi @StephenOTT, 
If I understand correctly, you want to target multiple fields (using a wildcard selecting them) in your query predicates and the entity will match if some of them matches

**Reply by StephenOTT** [reply]
Yes I am looking at ways to store dynamic data (such as user defined json).  I was looking at flattening the json into key/value. 

So if I had an array of values it would be: prop1.subProp[2].subSub1

Was looking if there was ~tricks to handle this where you could wildcard the array position. 

But it would seem I would need to store the paths as values instead of prop keys. Which further complicates querying and aggregations etc.

**Reply by StephenOTT** [reply]
We really need a way to store json and query it with ~ickle without proto configs :).

**Reply by fax4ever** [reply]
At the moment we don't suppor dynamic fields. Sorry
We're debating about the idea of eventually supporting them in future versions of the projects.
I think that eventually we will do it, since also Langchain4J (c.c. @karesti / @yrodiere) seems to require it.

**Reply by StephenOTT** [reply]
@fax4ever thanks for the quick feedback. Is there a material difference between the ickle query and running a stream filter?  I was considering just building a json object/array similar to a ES query or mongo filter/aggregation that converts into a Java stream and returns the result.  Any blockers that prevent this?
